apps/omg_eth/lib/omg_eth/application.ex

```diff
@@ -39,4 +42,25 @@ defmodule OMG.Eth.Application do
       {:error, :already_exists} -> :ok
     end
   end
+
+  defp valid_contracts() do
+    contracts_hash =
+      Configuration.contracts()
+      |> Map.put(:txhash_contract, Configuration.txhash_contract())
+      |> Map.put(:authority_addr, Configuration.authority_addr())
+      |> :erlang.phash2()
+
+    case DB.get_single_value(:omg_eth_contracts) do
+      result when result == :not_found or result == {:ok, 0} ->
+        multi_update = [{:put, :omg_eth_contracts, contracts_hash}]
+        :ok == DB.multi_update(multi_update)
+
+      {:ok, ^contracts_hash} ->
+        true
+
+      _ ->
+        _ = Logger.error("Contract addresses have changed since last boot!")
+        exit(:contracts_missmatch)
```

Comment:
Can we document the meaning of this failure? IMHO not sure where it should go ;)

Reply:
hmmm... is the error log above not sufficient?

Reply:
:face_palm: nvm